---
title: "crab-edgeR-test"
output: html_document
---
Rmd to test `edgeR` on 2019 crab RNAseq data.
```{r}
library(edgeR)
```
Files needed:
1. "targets_2019.txt" that contains treatment and sample information
2. "raw data" that is comprised of annotated BLAST output with gene IDs and counts (file made by Sam/Steven)

"targets_2019.txt" file was created based on manual for edgeR, what was done for EIMD 2019, and the data available in the 2019 crab seq. dataset
number of columns needs to be same as number count columns in rawdata_2019
Note: "L" = Day 26, stands for 'later'
      "E" = Day 12, stands for 'earlier'
      "U" = uninfected
      "I" = infected
I don't think using 12 and 26 works for R
upload file:
```{r}
targets_2019 <- readTargets("../analyses/targets_2019.txt")
targets_2019
```
set groups equal to the different samples
```{r}
group_2019 <- targets_2019$Day_InfectionStatus
group_2019 <- as.factor(group_2019)
group_2019
```

import raw count data 
```{r}
rawdata_2019 <- read.delim("https://raw.githubusercontent.com/sr320/nb-2019/master/C_bairdi/analyses/Abundance-merge.txt", header = TRUE)
head(rawdata_2019)
```
Make list of differentially expressed genes (DGE)
```{r}
y_2019 <- DGEList(counts = rawdata_2019[,2:9], group = group_2019)
head(y_2019$counts)
```
filtering out lowly expressed genes; from EIMD: 
"since the smallest group size is four, we keeping genes with at least one count per million in at least four samples"
I need to figure out how much I should filter out, but for now am trying same filtering that was done with EIMD nonzostera genes
```{r}
keep_2019 <- rowSums(cpm(y_2019)>1) >= 3
y_2019 <- y_2019[keep_2019,]
dim(y_2019) 
```
Keeping 56,039 genes in 8 libraries

recompute the library sizes: (this didn't work in nonZostera R, and isn't working here. i think it's because there are two $ in the first phrase)
```{r}
y_2019$samples$lib.size <- colSums(y_2019$counts)
```

# Calculating normalization factors
```{r}
y_2019 <- calcNormFactors(y_2019)
y_2019$samples 
```
From section 2.7 in manual: "a normalization factor below 1 indicates that a small number of high count genes are monopolizing the sequencing, causing the counts for other genes to be lower than would be usual given the library size. As a result, the library size will be scaled down, analogous to scaling the counts upwards in that library. Conversely, a factor above 1 scales up the library size, analogous to downscaling the counts"
These numbers are pretty similar to those from EIMD.. not too high above 1, not too low below 1... and those were deemed "good"


store lib.size in vector called "n" ... not sure why at the moment... 
```{r}
n <- y_2019$samples$lib.size
```

# Examining sample for outliers
Section 4.1.5 in manual "Data exploration"
Used same code as from EIMD... not sure if it works for crabs - will explore more 
```{r}
points <- c(0,1,2,3)
colors <- (rep(c("blue","red"),2))
plotMDS(y_2019, col=colors[group_2019], pch=points[group_2019])  
legend ("top", legend=levels(group_2019), pch=points, col=colors, ncol=2)
```
Section 3.2.3 of manual: "GLM Approach"
```{r}
design <-model.matrix(~0+group_2019)
colnames(design) <- levels(group_2019)
design
```
Section 2.10 of manual: "More complex experiemnts (glm functionality)
```{r}
y_2019 <- estimateGLMCommonDisp(y_2019, design, verbose=TRUE)
y_2019 <- estimateGLMTrendedDisp(y_2019,design)
y_2019 <- estimateGLMTagwiseDisp(y_2019,design)
plotBCV(y_2019)
```
Section 3.2.5 of manual: "A more traditional glm approach"
```{r}
fit <- glmQLFit(y_2019, design)
design
colnames(fit)
```

Estimating dispersion (pg 55 of manual)
Section 4.2.7 "Estimating the dispersion"
```{r}
y_2019 <- estimateDisp(y_2019, design, robust=TRUE)
```

```{r}
y_2019$common.dispersion
```
Value of common dispersion = 0.1399804. Dispersion is "estimating the genewise dispersion over all genes, allowing for possible abundance trend. The estimation is also robustified against potential outlier genes." 

```{r}
plotBCV(y_2019)
```
BCV = coefficient of biological variation - is the square root of dispersion
Therefore, the BCV = 0.37419546
Trended dispersion (blue) "shows a decreasing trend with expression level"

# Defining contrast groupings
Section 3.3.1 "Defining each treatment combination as a group"
Contrasts: What genes are DE in crabs exposed to Hamtodinium vs. controls?
The data file "targets_2019.txt" outlines the treatment conditions applied to each sample. 
```{r}
targets_2019
```
Make the different contrast options: 
Day 26 --> "L" for later
Day 12 --> "E" for earlier
"U" --> uninfected
"I" --> Infected
```{r}
my.contrasts <-makeContrasts(
  LU.LI=L_U-L_I, LU.EU=L_U-E_U, EU.EI=E_U-E_I, EI.LI=E_I-L_I, levels=design)
```

```{r}
qlf.LU.LI <- glmQLFTest(fit, contrast=my.contrasts[,"LU.LI"])
topTags(qlf.LU.LI)
qlf.LU.EU <- glmQLFTest(fit, contrast=my.contrasts[,"LU.EU"])
topTags(qlf.LU.EU)
qlf.EU.EI <- glmQLFTest(fit, contrast=my.contrasts[,"EU.EI"])
topTags(qlf.EU.EI)
qlf.EI.LI <- glmQLFTest(fit, contrast=my.contrasts[,"EI.LI"])
topTags(qlf.EI.LI)

summary(decideTests(qlf.LU.LI))
#Down          35212
#NotSig        16093
#Up             4734
summary(decideTests(qlf.LU.EU))
#Down          14418
#NotSig        40006
#Up             1615
summary(decideTests(qlf.EU.EI))
#Down          26340
#NotSig        20440
#Up             9259
summary(decideTests(qlf.EI.LI))
#Down           1721
#NotSig        53813
#Up              505

plotMD(qlf.LU.LI)
abline(h=c(-2,2), col="green")

plotMD(qlf.LU.EU)
abline(h=c(-2,2), col="green")

plotMD(qlf.EU.EI)
abline(h=c(-2,2), col="green")

plotMD(qlf.EI.LI)
abline(h=c(-2,2), col="green")
```



















