---
title: "prep-for-GO-MWU"
output: github_document
---
Script for formatting files for GO-MWU. 

GO-MWU needs two files comprised of all inputs (not just the DEGs):
1. tab-delimited file: two columns - gene IDs and GO terms separated by semicolons
Just need transcriptome BLAST output --> I have that! pull out gene ID and GO terms
2. .csv file: two columns - gene IDs and continuous measure of significance, I'll used padj. 
These measures of significance need to be associated with GO enrichment. 
Come from `DESeq` from SR --> project-crab/analyses/deseq2.res.tab

Again, both files need to be all data inputs, not just the DEGs. 
https://github.com/z0on/GO_MWU

```{r}
library(tidyverse)
```

## Read in files:
Read in counts file from SR - 2019 crab dataset: 
5 columns: Trinity_ID (need to add column name), and 4 sample names
```{r}
crab_counts <- read.delim("../analyses/kallisto-0315.isoform.counts.matrix", header = TRUE)
head(crab_counts)
```
Add name to first column: "Trinity_ID"
```{r}
colnames(crab_counts) <- c("Trinity_ID", "329774_out", "329775_out", "329776_out", "329777_out")
head(crab_counts)
```
Good to go.

Import uniprot GO file downloaded from uniprot.org 
(has GO IDs, GOBP, GOCC, and GO MF --> file too big to be on Github, but it's all the data and is saved locally in my project-crab repo)
```{r}
go <- read.delim("../data/uniprot-reviewed.tab", sep = '\t', header = TRUE)
head(go)
```

## Create GO annotations table
Join `go` and `cr






