---
title: "prep-for-GO-MWU"
output: github_document
---
Script for formatting files for GO-MWU. 

GO-MWU needs two files comprised of all inputs (not just the DEGs):
1. tab-delimited file: two columns - gene IDs and GO terms separated by semicolons
Just need transcriptome BLAST output --> I have that! pull out gene ID and GO terms
2. .csv file: two columns - gene IDs and continuous measure of significance, I'll used padj. 
These measures of significance need to be associated with GO enrichment. 
Come from `DESeq` from SR --> project-crab/analyses/deseq2.res.tab

Again, both files need to be all data inputs, not just the DEGs. 
https://github.com/z0on/GO_MWU

```{r}
library(tidyverse)
```

## Read in files:
Read in counts file from SR - 2019 crab dataset: 
5 columns: Trinity_ID (need to add column name), and 4 sample names
```{r}
crab_counts <- read.delim("../analyses/kallisto-0315.isoform.counts.matrix", header = TRUE)
head(crab_counts)
```
Add name to first column: "Trinity_ID"
```{r}
colnames(crab_counts) <- c("Trinity_ID", "329774_out", "329775_out", "329776_out", "329777_out")
head(crab_counts)
```
Good to go.

Import BLAST output file from crab transcriptome:
(I pasted into excel to get the data into the correct columns...)
```{r}
blastout <- read.delim("../analyses/crab-blastout.txt", header = FALSE)
head(blastout)
```
Rename column "V3" --> "Entry" so that it matches the column name rom the uniprot ID list; also rename column "V1" --> "Trinity_ID"
```{r}
colnames(blastout) <- c("Trinity_ID", "V2", "Entry", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blastout)
```

Import uniprot GO file downloaded from uniprot.org 
(has GO IDs, GOBP, GOCC, and GO MF --> file too big to be on Github, but it's all the data and is saved locally in my project-crab repo)
```{r}
go <- read.delim("../data/uniprot-reviewed.tab", sep = '\t', header = TRUE)
head(go)
```

## Create GO annotations table
Join `go` and `blastout` to annotate the crab transcriptome BLAST output with GO terms:
```{r}
annot_blast <- left_join(blastout, go, by = "Entry")
head(annot_blast)
```

Create crab GO annotations .tab by pulling out Trinity_ID column along with "Gene.ontology.IDs" column: 
```{r}
crab_GO_annot <- subset(annot_blast, select = c(Trinity_ID, Gene.ontology.IDs))
head(crab_GO_annot)
```
Write out the file to analyses/GO-MWU, making sure it's tab-delimited (remove row names)
```{r}
write.table(crab_GO_annot, "../analyses/GO-MWU/crab-GO-annot.tab", sep = "\t", row.names = FALSE, quote = FALSE, col.names = FALSE)
```

## Create Table of significance measures
(basing off of script from EIMD: https://github.com/eimd-2019/project-EWD-transcriptomics/blob/master/analyses/GO-MWU/2019-07-11-Gene-Enrichment-with-GO-MWU.Rmd)

read in file that SR made with DESeq that includes ALL the 2019 crab dataset, not just the differentially expressed genes:
```{r}
crabgenes <- read.table("../analyses/deseq2.res.tab", sep = "\t", header = TRUE)
head(crabgenes)
```
Add name for first column with Trinity IDs is technically the row.names, so I can't rename the first column - it's not a real column.

Make the rownames into the first column: 
```{r}
crabgenes <- tibble::rownames_to_column(crabgenes, "Trinity_ID")
head(crabgenes)
```

Attempt to create table of sig. measures by subsetting row names and adjusted p-values (padj)
```{r}
crab_all_pval <- subset(crabgenes, select = c(Trinity_ID, pvalue))
head(crab_all_pval)
```
Write out as .csv

```{r}
write.csv(crab_all_pval, "../analyses/GO-MWU/2019-crab-ALL-geneID-pval.csv", row.names = FALSE, quote = FALSE)
```





