---
title: "trying-edgeR"
output: html_document
---
trying to replicate class script with edgeR
then will try on crab stuff

EdgeR - differential expression analysis of transcriptomic data
try using script for nonzostera diff. exp. 
https://github.com/eimd-2019/project-EWD-transcriptomics/blob/master/analyses/EdgeR/GRASS_START_GENE_MEE_Contrasts_CAB_7_10_nonZostera_at.R

```{r}
library(edgeR)
```

make a matrix of factors called "targets", Treat=Pathogen, Heat=Temp
got file in an email from Colleen
```{r}
targets <-readTargets("../../../../Desktop/MasterWork/edgeR-practice/EIMD-stuff/targets.txt")
targets
```

set groups equal to time differences
```{r}
group <- targets$Treat_Heat
group <- as.factor(group)
group
```
import raw data
```{r}
rawdata <- read.delim("../../project-EWD-transcriptomics/data/nonZostera-blast-annot-withGeneID-noIsoforms-geneCounts.tab", header = FALSE)
  #read.delim("nonZostera-blast-annot-withGeneID-noIsoforms-geneCounts.tab", header=FALSE)
head(rawdata)
colnames(rawdata)<- c("GeneID", "Accession", "Isoform", "E-value", "ProteinN", "GO_BP", "GO_CC", "GO_MF", "GO", "Status", "Organism", "S_10B", "S_9A", "S_13A", "S_42A", "S_46B", "S_47B", "S_48B", "S_2A", "S_2B", "S_7B", "S_8B", "S_33A", "S_36B", "S_38A", "S_40A")
row.names(rawdata)<-rawdata$GrassGene
head(rawdata)
```
Make DGE list
```{r}
y <- DGEList(counts=rawdata[,12:26],group=group)
head(y$counts)
```

filtering out lowly expressed genes; since the smallest group size is four, we keeping genes with at least one count per million in at least four samples
```{r}
keep <- rowSums(cpm(y)>1) >= 3
y <- y[keep,]
dim(y) 
```
Keeping 1281 genes in 15 libraries from Zostera file

recompute the library sizes:
```{r}
y$samples$lib.size <- colSums(y$counts)
```

2.7
calculating normalization factors
```{r}
y <- calcNormFactors(y)
y$samples 
```
looks good

```{r}
n <- y$samples$lib.size
```

examine sample for outliers
4.15
```{r}
points <- c(0,1,2,3)
colors <- (rep(c("blue","red"),2))
plotMDS(y, col=colors[group], pch=points[group])  
legend ("top", legend=levels(group), pch=points, col=colors, ncol=2)
```

3.23
```{r}
design <-model.matrix(~0+group)
colnames(design) <- levels(group)
design
```

2.10
```{r}
y <- estimateGLMCommonDisp(y, design, verbose=TRUE)
y <- estimateGLMTrendedDisp(y,design)
y <- estimateGLMTagwiseDisp(y,design)
plotBCV(y)
```

3.25
```{r}
fit <- glmQLFit(y, design)
design
colnames(fit)
```

Estimating dispersion (pg 54 of manual)
```{r}
#y <- estimateDisp(y, design, robust=TRUE)
```
Install statmod
```{r}
install.packages("statmod")
```
retry estimating dispersion (pg 54)
```{r}
y <- estimateDisp(y, design, robust=TRUE)
```

```{r}
y$common.dispersion
```

```{r}
plotBCV(y)
```
page 48 --> coeeficient of biological variation = 1.36

Set up fpllowing 3.31 (from older level, different order)
MEE contrasts: What genes are DE in shoots exposed to laby vs controls? Make 4 contrasts and look for genes DE in all 4
```{r}
my.contrasts <-makeContrasts(
  EXPH.CONH=EXP_HOT-CON_HOT, EXPH.CONC=EXP_HOT-CON_COLD, EXPC.CONH=EXP_COLD-CON_HOT, EXPC.CONC=EXP_COLD-CON_COLD, levels=design)

qlf.EXPH.CONH <- glmQLFTest(fit, contrast=my.contrasts[,"EXPH.CONH"])
topTags(qlf.EXPH.CONH)
qlf.EXPH.CONC <- glmQLFTest(fit, contrast=my.contrasts[,"EXPH.CONC"])
topTags(qlf.EXPH.CONC)
qlf.EXPC.CONH <- glmQLFTest(fit, contrast=my.contrasts[,"EXPC.CONH"])
topTags(qlf.EXPC.CONH)
qlf.EXPC.CONC <- glmQLFTest(fit, contrast=my.contrasts[,"EXPC.CONC"])
topTags(qlf.EXPC.CONC)

summary(decideTests(qlf.EXPH.CONH))
#UP: 56, DOWN:2
summary(decideTests(qlf.EXPH.CONC))
#UP: 56, DOWN: 3
summary(decideTests(qlf.EXPC.CONH))
#UP: 61, DOWN: 22
summary(decideTests(qlf.EXPC.CONC))
#UP: 57, DOWN: 10

plotMD(qlf.EXPH.CONH)
abline(h=c(-2,2), col="green")

plotMD(qlf.EXPH.CONC)
abline(h=c(-2,2), col="green")

plotMD(qlf.EXPC.CONH)
abline(h=c(-2,2), col="green")

plotMD(qlf.EXPC.CONC)
abline(h=c(-2,2), col="green")
```

MEE contrasts: what genes are DE between laby exposed shoots hot and cold treatments?
```{r}
my.contrasts <-makeContrasts(
  EXPH.EXPC=EXP_HOT-EXP_COLD, levels=design)
qlf.EXPH.EXPC <- glmQLFTest(fit, contrast=my.contrasts[,"EXPH.EXPC"])
topTags(qlf.EXPH.EXPC)

summary(decideTests(qlf.EXPH.EXPC))
#UP: 0, DOWN: 0
```

MEE contrasts: what genes are DDE between hot and cold control shoots?
```{r}
my.contrasts <-makeContrasts(
  CONH.CONC=CON_HOT-CON_COLD, levels=design)
qlf.CONH.CONC <- glmQLFTest(fit, contrast=my.contrasts[,"CONH.CONC"])
topTags(qlf.CONH.CONC)

summary(decideTests(qlf.CONH.CONC))
#0

```

MEE contrasts: what genes are DE between exposed (hot and cold) and  control (hot and cold) shoots?
```{r}
my.contrasts4 <-makeContrasts(
  EXP.CON=(EXP_HOT+EXP_COLD)-(CON_HOT+CON_COLD), levels=design)
 qlf.EXP.CON <- glmQLFTest(fit, contrast=my.contrasts4[,"EXP.CON"])
topTags(qlf.EXP.CON)

summary(decideTests(qlf.EXP.CON))
#UP: 79, DOWN: 21
```

looking at the counts per million
```{r}
top <- rownames(topTags(qlf.EXP.CON))
cpm(y)[top,]
```

MEE Graphs of 3 more contrasts listed above: 

```{r}
plotMD(qlf.EXPH.EXPC)
abline(h=c(-2,2), col="black")

plotMD(qlf.CONH.CONC)
abline(h=c(-2,2), col="black")

plotMD(qlf.EXP.CON)
abline(h=c(-1,1), col="blue")
abline(h=c(-5,5), col="green")
abline(h=c(-10,10), col="purple")

logcpm <- cpm(y, log=TRUE)
```

the following is said for us to ignore for now... not sure why:
```{r}
#ignore this for now :)

#Writetable
#Inf=infinite number so all samples
top <-topTags(qlf.EXP.CON, n=Inf)
write.table (top, "EXP.CON.txt", sep="\t")

#Smaller list
#FDR correct p-values, 0.05
out <- topTags(qlf.EXP.CON, n=Inf, adjust.method="BH")
keep <- out$table$FDR <= 0.05 
out[keep,]
write.table(out$table[keep,], file="DE.EXP.CON.FDR.nZ.txt", row.names = TRUE, col.names = NA, quote=FALSE, sep = "\t")
```

