---
title: "2019-crab-infection-GO-MWU_BP"
output: html_document
---
Rmd for using `GO-MWU` with the 2019 crab dataset. 
Modelled after the GO-MWU.R script from the GO-MWU github repo:
https://github.com/z0on/GO_MWU/blob/master/GO_MWU.R 

```{r}
library(ape)
library(tidyverse)
library(dplyr)
```

Make sure the working directory is the directory: project-crab/analyses/GO-MWU: 
```{r}
getwd()
```
Good to go. 

Two crab data files will be used.
1. ("input") is a .csv file that is comprised of 2 columns. The data is from ALL of the 2019 crab dataset `DESeq` results, not just the DEGs (performed by SR). Column 1 is the gene IDs, column 2 is a continuos measure of significance. In this anaylsis, I'll use the adjusted p-values.
2. ("goAnnotations") is a .tab file that is also comprised of 2 columns. The data are from the crab transcriptome, which is made of all crab RNAseq datasets. Column 1 are the gene IDs, column 2 are the GO terms, separated by ';'. 

I will be using the GO Biological Process division. 

## Add binary measure to the input file (in order to base GO enrichment on Fisher's Exact test) 
0 or 1 
0 for not significant
1 for significant

Load the 2019-crab-ALL-geneID-padj.csv file:
```{r}
crab_padj <- read_csv("2019-crab-ALL-geneID-padj.csv")
head(crab_padj)
```

If the p-adj is less than or equal to 0.05, it is significant (assign a 1). If p-adj is greater than 0.05, it is not significant (assign a 0). 
There should be 772 significan genes (SR found 772 DEGs with `DESeq`)

create a new column called "binary_sig"
```{r}
crab_padj_binary <- crab_padj %>%
  mutate(binary_sig = if_else(crab_padj$padj <= 0.05, "1", "0"))
head(crab_padj_binary)
```

Create new input file that pulls out the Trinity_ID and the binary_sig columns:
```{r}
crabALL_binary_sig <- subset(crab_padj_binary, select = c(Trinity_ID, binary_sig))
head(crabALL_binary_sig)
```

write out as a .csv to use in GO-MWU below: 
```{r}
write.csv(crabALL_binary_sig, "2019-crab-ALL-geneID-binary.csv", quote = FALSE, row.names = FALSE)
```

## Run GO-MWU 
### Set variables
Use the binary .csv created above
From GO-MWU script: 
```{r}
#Edit these to match your data file names: 
input="2019-crab-ALL-geneID-binary.csv" # two columns of comma-separated values: gene id, continuous measure of significance. To perform standard GO enrichment analysis based on Fisher's exact test, use binary measure (0 or 1, i.e., either sgnificant or not).
goAnnotations="crab-GO-annot.tab" # two-column, tab-delimited, one line per gene, multiple GO terms separated by semicolon. If you have multiple lines per gene, use nrify_GOtable.pl prior to running this script.
goDatabase="go.obo" # download from http://www.geneontology.org/GO.downloads.ontology.shtml
goDivision="MF" # either MF, or BP, or CC
source("gomwu.functions.R")
```

### Calculating results (from GO-MWU script):
```{r}
# Calculating stats. It might take ~3 min for MF and BP. Do not rerun it if you just want to replot the data with different cutoffs, go straight to gomwuPlot. If you change any of the numeric values below, delete the files that were generated in previos runs first.
gomwuStats(input, goDatabase, goAnnotations, goDivision,
           perlPath="../../../../../../../usr/bin/perl", # replace with full path to perl executable if it is not in your system's PATH already
           largest=0.1,  # a GO category will not be considered if it contains more than this fraction of the total number of genes
           smallest=5,   # a GO category should contain at least this many genes to be considered
           clusterCutHeight=0.25, # threshold for merging similar (gene-sharing) terms. See README for details.
           #	Alternative="g" # by default the MWU test is two-tailed; specify "g" or "l" of you want to test for "greater" or "less" instead. 
           #	Module=TRUE,Alternative="g" # un-remark this if you are analyzing a SIGNED WGCNA module (values: 0 for not in module genes, kME for in-module genes). In the call to gomwuPlot below, specify absValue=0.001 (count number of "good genes" that fall into the module)
           #	Module=TRUE # un-remark this if you are analyzing an UNSIGNED WGCNA module 
)
# do not continue if the printout shows that no GO terms pass 10% FDR.
```
Found 2 GO terms.... 
Won't plot, but I'll look at results: 
```{r}
results
```

GAH! still getting 0.... 
